snapm/_progress.py

```
@@ -0,0 +1,620 @@
+                      the progress bar will occupy this width. Cannot be used
+                      with ``width_frac``.
+        :type width: ``int``
+        :param width_frac: An optional width fraction between [0..1]. If
+                           specified the terminal width is multiplied by this
+                           value and rounded to the nearest integer to
+                           determine the width of the progress bar. Cannot be
+                           used with ``width``.
+        :type width_frac: ``Optional[float]``
+        """
+        if width is not None and width_frac is not None:
+            raise ValueError("width and width_frac are mutually exclusive")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)